runtime-sdk/src/modules/core/types.rs

```diff
@@ -32,3 +32,37 @@ pub struct CallDataPublicKeyQueryResponse {
     /// Public key used for deriving the shared secret for encrypting call data.
     pub public_key: SignedPublicKey,
 }
+
+#[derive(Debug, Copy, Clone, cbor::Encode, cbor::Decode)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub enum MethodHandlerKind {
+    Call,
+    // `Prefetch` is omitted because it is an implementation detail of handling `Call`s.
+    Query,
+    MessageResult,
+}
+
+#[derive(Debug, Clone, cbor::Encode, cbor::Decode)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct MethodHandlerInfo {
+    pub kind: MethodHandlerKind,
+    pub name: String,
+}
+
+/// Metadata for an individual module.
+#[derive(Clone, Debug, cbor::Encode, cbor::Decode)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct ModuleInfo {
+    pub version: u32,
+    pub params: Vec<u8>,
+    pub methods: Option<Vec<MethodHandlerInfo>>,
```

Comment:
Why not just a vector that can be empty? Any special meaning for `None`?

Reply:
This field reflects the return value of https://github.com/oasisprotocol/oasis-sdk/pull/758/files#diff-b038871be26b907a37aa2a44827ccaa92557482f5cffcca211fbff23de45523dR194: `A return value of None means that the implementor does not care to list the methods, or the implementor is a tuple of modules.`

The idea was to distinguish between a module that explicitly has no methods vs a module that just doesn't support reflection (e.g. because it's not using `sdk_derive()`. On second thought, a module with no methods makes little sense. Lemme simplify the type to just a `Vec`.

Reply:
Done